### ISSUE TITLE
feat(inputs / outputs): output from signal

### ISSUE THREAD

**Issue by mauriziocescon** [angular/angular#57943]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

Hey! 

As part of the ongoing discussion regarding the usage of effects, I guess we might miss something that would be really handy: outputs from signals (declarative way). 

This a variation of the `model` pattern (which is great) that IMHO naturally complements the potential introduction of a `wComputed` (see https://github.com/angular/angular/issues/55673). Here is a concrete example of what I mean (albeit very simple and refactorable): 

```ts 
@Component({
  selector: 'app-text',
  standalone: true,
  imports: [FormsModule],
  template: `
    <div>Type some value:</div>
    <input type="text" [(ngModel)]="value"/>
  `,
})
export class TextComponent {
  value = signal<string>('');

  private length = computed(() => this.value().length);
  private length$ = toObservable(this.length);
  lengthDidChange = outputFromObservable(this.length$);

  /**
   * I'd like to emit any time lenght changes.
   *
   * Ideally:
   *
   * lengthDidChange = output<number>({ from: this.length });
   * or
   * lengthDidChange = output<number>({ from: ()=> {return ... } });
   * or
   * lengthDidChange = outputFrom<number>(()=> {return ... });
   */
}

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [TextComponent],
  template: `
    <app-text (lengthDidChange)="lengthDidChange($event)"/>
    <hr>
    <div>{{ desc() }}</div>
  `,
})
export class App {
  length = signal<number | undefined>(undefined);
  desc = computed(() => `Length is ${this.length()}`);

  lengthDidChange(value: number): void {
    this.length.set(value);
  }
}
```

https://stackblitz.com/edit/stackblitz-starters-lazelp?file=src%2Fmain.ts

The pattern is basically: any time a component `internal` signal changes => emit the new value (no need to call emit, no observables, ...). 

Of course, you could say: in the example above, you should use `ngModelChange`, setup a method in `TextComponent ` and emit. That's true for such example (again very simple). 

But as usual, the story changes as soon as the example gets more complicated (more components, more needs to synchronise values, ...). In general, I think more complicated scenarios could be addressed by devs using `effects` (which is by the way the case with `toObservable` 😅). 

I hope I haven't missed anything trivial (stupidity is always around the corner 😅). 

Thanks a lot! 

### Proposed solution

Introduce some sort of declarative way of creating an `output` from signals. 

### Alternatives considered

Depends on the use case. In general, the way to go is to have a more imperative approach to `emit`.

**Comment by JeanMeche** [comment]
Hi, this topic has been brought up in the past in #57208 and #56923. 

TLDR, Signals are states, outputs are events.

**Comment by mauriziocescon** [comment]
Ah, thanks! Missed those guys!

**Comment by mauriziocescon** [comment]
Maybe a clear mentioned in the doc might help: it's not really immediate to understand the difference between a `model` and a `declarative output`. 

https://angular.dev/guide/signals/model#implicit-change-events

**Comment by mauriziocescon** [comment]
Ok, I checked the code for [`model`](https://github.com/angular/angular/blob/main/packages/core/src/authoring/model/model_signal.ts#L78): `set` / `update` are intercepted (sort of proxy)! Which makes sense for a `wSignal`, but probably not for a `computed`. 

So, I guess: an `outputFromWritableSignal` would be possible while an `outputFromComputed` should fallback to an `effect` with a conceptual fork of the logic. So, a sort of potential mess for developers (why are they working differently? 😅). 

Anyway, thanks again for the feedback! 


https://stackblitz.com/edit/stackblitz-starters-3hoxbx?file=src%2Fmain.ts

<img width="1254" alt="Screenshot 2024-09-25 at 10 30 17" src="https://github.com/user-attachments/assets/e34d574e-a938-4f0c-91e4-32abb9472742">
